contracts/SpokePool.sol

```diff
@@ -38,11 +44,11 @@ abstract contract SpokePool is SpokePoolInterface, Testable, Lockable, MultiCall
 
     // Address of wrappedNativeToken contract for this network. If an origin token matches this, then the caller can
     // optionally instruct this contract to wrap native tokens when depositing (ie ETH->WETH or MATIC->WMATIC).
-    WETH9 public immutable wrappedNativeToken;
+    WETH9 public wrappedNativeToken;
 
     // Any deposit quote times greater than or less than this value to the current contract time is blocked. Forces
     // caller to use an approximately "current" realized fee. Defaults to 10 minutes.
-    uint32 public depositQuoteTimeBuffer = 600;
+    uint32 public depositQuoteTimeBuffer;
```

Comment:
It [won't be possible to add any more state variables](https://docs.openzeppelin.com/upgrades-plugins/1.x/writing-upgradeable#modifying-your-contracts) to the base SpokePool after deployment. Have we ruled out adding some placeholders, just in case we find a use for something else in future?

Reply:
Great point https://github.com/across-protocol/contracts-v2/pull/218/commits/6a9f5093193229800f4e46aaa52dd1d54e3fe667

What do you and @mrice32 think about adding 50 storage slots for future upgrades?

Reply:
Great call! Any reason not to make it way way bigger than we'd ever need? Like 1000? Just to be super safe

Reply:
Here's a thought experiment. Imagine we wanted to support bridging with arbitrary data so that someone could bridge-and-swap. Say we had to upgrade the `deposit()` function (or add a new depositWithData function). Would this be possible with upgradeable proxies? How much of the extra storage slots would we need?

Reply:
It depends on the implementation. If the data is just hashed into the deposit data and added to an event, no storage changes would be necessary.

If we needed to store extra info about each deposit, deposits would need extra storage, but they aren't stored continuously, so that can be done without pre-reserving slots.

In this implementation, what extra contiguous storage would you imagine needing?

Reply:
I'm not sure, this was just a functionality that I discussed with Ryan about adding and wanted to make sure we're prepared for it. Basically bridge-and-swap would be a good UX addition

Reply:
What if we needed to add a new Base contract function, does that take up slots?

Reply:
Nope, functions don't take up slots or anything. They are just more bytecode in the implementation contract.